site/content/entry/yuzu-progress-report-jun-2023/index.md

```diff
@@ -0,0 +1,738 @@
+Android versions prior to the upcoming 14 don’t let apps know the exact RAM amount of the device, so available free RAM is used instead.
+
+Speaking of which, t895 added {{< gh-hovercard "10945" "support for Android 14." >}} 
+Better be ready beforehand! 
+You never know when Google might surprise us.
+
+## Making CPUs go the opposite of BRRR, AMD edition
+
+[Back in March](https://yuzu-emu.org/entry/yuzu-progress-report-mar-2023/#making-cpus-go-the-opposite-of-brrrr), we explained why Windows is not accurate enough when very high precision timers are required, so in order to improve performance, power consumption, and temperatures on x86_64 CPUs, specific CPU instructions are needed to reduce the time the CPU spends improperly idling.
+We have a big mistake to correct here. We said only Intel offered such instructions starting with Alder Lake, and that AMD didn’t offer any.
+
+Fortunately, we were wrong! AMD does indeed have its own implementation, the `monitorx` and `mwaitx` instructions pair, which have been out since *2015*, predating Ryzen!
+
+By doing his usual black magic, [Morph](https://github.com/Morph1984) implemented support for these instructions, providing the same benefits Intel users got to previously enjoy alone, namely performance and battery life improvements for power-limited products (laptops, handhelds) or CPUs with few core counts.
+
+Since AMD CPUs running Windows can now {{< gh-hovercard "10935" "properly idle for longer" >}} while waiting, power consumption and/or performance numbers are improved, depending on which limit is hit first, the power budget of the CPU, or a framerate cap.
+Sadly, your writer (who got tasked with testing this change) doesn’t have access to an Asus Rog Ally handheld, or even a Ryzen laptop, to do proper benchmarks, so instead please accept the numbers collected with a desktop Ryzen 5 5600X.
+The results should be better with a proper mobile platform, but alas, we can only show what we tested.
+
+Power consumption got a 10-16% reduction, which translates with this particular CPU model to around a 10W reduction.
+While simulating a power limited scenario (capping the PPT to 40W), performance improved up to 60%, but on average a 20% gain is more often expected.
+
+Now we can guess what some of you may be thinking, “how much does this help the Steam Deck? That’s a power-limited system with only four cores!” 
+It makes no difference on the Deck. Let us explain.
+
+The Steam Deck by default runs a Linux distribution. 
+Linux doesn’t benefit from these changes as its kernel, unlike Windows’, is actually decent and can run high-precision timer events without any problem or undocumented SDK usage. If there’s no need to spin the CPU at all, it gets to idle properly. 
+So unless you installed Windows onto your Steam Deck, you already had the best performance it can offer.
+
+Long gameplay sessions can sometimes feel like a gamble regarding stability. 
```

Comment:
random linux section stuck in the middle of an unrelated header

Reply:
Making a kernel section for just this would be a bit sad innit

Reply:
All I'm saying is, it definitely doesn't belong here. Maybe a new section is or isn't warranted, but this feels extremely out of place

Reply:
Let me see what I can cook